src/main/java/smokefree/domain/Initiative.java

```diff
@@ -206,5 +216,46 @@ private void assertCurrentUserIsManager(MetaData metaData) {
                 "You are not a manager of this playground"));
     }
 
+    private void validateMaximumPlaygroundCapacity() {
+        if(initiativeProjection.playgrounds().size() >= initiativeProjection.MAXIMUM_PLAYGROUNDS_ALLOWED) {
+            throw new DomainException("MAXIMUM_PLAYGROUNDS_CAPACITY_REACHED",
+                    "Can not add more than " + initiativeProjection.MAXIMUM_PLAYGROUNDS_ALLOWED + " playgrounds",
+                    "Sorry, Maximum playgrounds capacity is reached, please contact helpline");
+        }
+    }
+
+    private void validateDuplicatePlaygroundNames(String playgroundName) {
+        initiativeProjection.playgrounds().entrySet()
+                .stream()
+                .filter( playgroundEntry -> playgroundEntry.getValue().getName().equals(playgroundName))
+                .map(playgroundEntry -> playgroundEntry.getValue())
+                .findFirst()
+                .ifPresent( p -> {
+                    throw new DomainException("DUPLICATE_PLAYGROUND_NAME",
+                            "Playground " + playgroundName + " does already exist, please choose a different name",
+                            "Playground name does already exist");
+                });
+    }
+
+    private void validatePlaygroundsRange(GeoLocation newPlaygroundLocation, long distance) {
+        GeodeticCalculator geodeticCalculator = new GeodeticCalculator();
+        Ellipsoid ellipsoidsReference = Ellipsoid.WGS84;
+        GlobalPosition newPlaygroundPosition = new GlobalPosition(newPlaygroundLocation.getLat(), newPlaygroundLocation.getLng(), 0.0); // Point A
+        initiativeProjection.playgrounds().entrySet().stream()
+                .filter( playgroundEntry -> {
+                    Playground playground = playgroundEntry.getValue();
+                    GlobalPosition currentPlaygroundPosition = new GlobalPosition(playground.getLat() , playground.getLng(), 0.0);
+                    double playgroundsDistance = geodeticCalculator.calculateGeodeticCurve(ellipsoidsReference, currentPlaygroundPosition, newPlaygroundPosition).getEllipsoidalDistance();
+                    return  playgroundsDistance < distance ? true:false;
+                }).map(playgroundEntry -> playgroundEntry.getValue())
+                .findFirst()
+                .ifPresent(p -> {
+                    throw new DomainException("PLAYGROUNS_LOCATED_CLOSELY",
+                            "Two playgrounds can not exist within 100 Meters",
```

Comment:
hmm 100 ? Something is wrong here. Please see for yourself what.

Reply:
In the earlier discussion, we agreed for 100 Meters. Anyway, it can be changed.

Reply:
Sorry, wrong answer :(
 Have a good look and try again.